packages/file_selector/file_selector_web/lib/file_selector_web.dart

```diff
@@ -0,0 +1,222 @@
+      _verifyXTypeGroup(group);
+
+      for (String mimeType in group.mimeTypes ?? []) {
+        allTypes.add(mimeType);
+      }
+      for (String extension in group.extensions ?? []) {   
+        String ext = extension;
+        if (ext.isNotEmpty && [0] != '.') {
+          ext = '.' + ext;
+        }
+        
+        allTypes.add(ext);
+      }
+      for (String webWildCard in group.webWildCards ?? []) {
+        allTypes.add(webWildCard);
+      }
+    }
+    return allTypes?.where((e) => e.isNotEmpty)?.join(',') ?? '';
+  }
+
+  /// Creates a file input element with only the accept attribute
+  @visibleForTesting
+  FileUploadInputElement createFileInputElement(String accepted, bool multiple) {
+    if (_hasTestOverrides && _overrides.createFileInputElement != null) {
+      return _overrides.createFileInputElement(accepted, multiple);
+    }
+    
+    final FileUploadInputElement element = FileUploadInputElement();
+    if (accepted.isNotEmpty) {
+      element.accept = accepted;
+    }
+    element.multiple = multiple;
+
+    return element;
+  }
+
+  void _addElementToDomAndClick(Element element) {
+    // Add the file input element and click it
+    // All previous elements will be removed before adding the new one
+    _target.children.clear();
+    _target.children.add(element);
+    element.click();
+  }
+
+  List<XFile> _getXFilesFromFiles (List<File> files) {
+    List<XFile> xFiles = List<XFile>();
+    
+    Duration timeZoneOffset = DateTime.now().timeZoneOffset;
+
+    for (File file in files) {
+      String url = Url.createObjectUrl(file);
+      String name = file.name;
+      int length = file.size;
+      DateTime modified = file.lastModifiedDate.add(timeZoneOffset);
+
+      xFiles.add(XFile(url, name: name, lastModified: modified));
+    }
+
+    return xFiles;
+  }
+
+  /// Getter for retrieving files from an input element
+  @visibleForTesting
+  List<File> getFilesFromInputElement(InputElement element) {
+    if(_hasTestOverrides && _overrides.getFilesFromInputElement != null) {
+      return _overrides.getFilesFromInputElement(element);
+    }
+
+    return element?.files ?? [];
+  }
+  
+  /// Listen for file input element to change and retrieve files when
+  /// this happens.
+  @visibleForTesting
+  Future<List<XFile>> getFilesWhenReady(InputElement element)  {
+    if(_hasTestOverrides && _overrides.getFilesWhenReady != null) {
+      return _overrides.getFilesWhenReady(element);
+    }
+
+    final Completer<List<XFile>> _completer = Completer();
+
+    // Listens for element change
+    element.onChange.first.then((event) {
+      // File type from dart:html class
+      final List<File> files = getFilesFromInputElement(element);
+
+      // Create XFile from dart:html Files
+      final xFiles = _getXFilesFromFiles(files);
+
+      _completer.complete(xFiles);
+    });
+
+    element.onError.first.then((event) {
+      if (!_completer.isCompleted) {
+        _completer.completeError(event);
```

Comment:
Instead of completing with the `event`, this probably should extract information from the error, and wrap it in a `PlatformException`.